internal/controller/pgupgrade/apply.go

```diff
@@ -0,0 +1,164 @@
+			"value": value,
+		}
+		return patch
+	}
+
+	*patch = append(*patch, f)
+
+	return f
+}
+
+// Remove appends a "remove" operation to patch.
+//
+// > The "remove" operation removes the value at the target location.
+// >
+// > The target location MUST exist for the operation to be successful.
+//
+
+func (patch *JSON6902) Remove(path ...string) *JSON6902 {
+	*patch = append(*patch, map[string]interface{}{
+		"op":   "remove",
+		"path": patch.pointer(path...),
+	})
+
+	return patch
+}
+
+// Replace appends a "replace" operation to patch.
+//
+// > The "replace" operation replaces the value at the target location
+// > with a new value.
+// >
+// > The target location MUST exist for the operation to be successful.
+//
+
+func (patch *JSON6902) Replace(path ...string) func(value interface{}) *JSON6902 {
+	i := len(*patch)
+	f := func(value interface{}) *JSON6902 {
+		(*patch)[i] = map[string]interface{}{
+			"op":    "replace",
+			"path":  patch.pointer(path...),
+			"value": value,
+		}
+		return patch
+	}
+
+	*patch = append(*patch, f)
+
+	return f
+}
+
+// Bytes returns the JSON representation of patch.
+func (patch JSON6902) Bytes() ([]byte, error) { return patch.Data(nil) }
+
+// Data returns the JSON representation of patch.
+func (patch JSON6902) Data(client.Object) ([]byte, error) { return json.Marshal(patch) }
+
+// IsEmpty returns true when patch has no operations.
+func (patch JSON6902) IsEmpty() bool { return len(patch) == 0 }
+
+// Type returns k8s.io/apimachinery/pkg/types.JSONPatchType.
+func (patch JSON6902) Type() types.PatchType { return types.JSONPatchType }
+
+// patch sends patch to object's endpoint in the Kubernetes API and updates
+// object with any returned content. The fieldManager is set to r.Owner, but
+// can be overridden in options.
+// - https://docs.k8s.io/reference/using-api/server-side-apply/#managers
+func (r *PGUpgradeReconciler) patch(
+	ctx context.Context, object client.Object,
+	patch client.Patch, options ...client.PatchOption,
+) error {
+	options = append([]client.PatchOption{r.Owner}, options...)
+	return r.Client.Patch(ctx, object, patch, options...)
+}
+
+// apply sends an apply patch to object's endpoint in the Kubernetes API and
+// updates object with any returned content. The fieldManager is set to
+// r.Owner and the force parameter is true.
+// - https://docs.k8s.io/reference/using-api/server-side-apply/#managers
+// - https://docs.k8s.io/reference/using-api/server-side-apply/#conflicts
+func (r *PGUpgradeReconciler) apply(ctx context.Context, object client.Object) error {
+	// Generate an apply-patch by comparing the object to its zero value.
+	zero := reflect.New(reflect.TypeOf(object).Elem()).Interface()
+	data, err := client.MergeFrom(zero.(client.Object)).Data(object)
+	apply := client.RawPatch(client.Apply.Type(), data)
+
+	// Send the apply-patch with force=true.
+	if err == nil {
+		err = r.patch(ctx, object, apply, client.ForceOwnership)
+	}
+
+	return err
+}
```

Comment:
This doesn't quite look like the `apply` that we have for either pgo or pgo-upgrade right now. Was there a decision made to change it?

Reply:
It should be identical to pg-upgrade...

Reply:
It was copied directly from `pgo-upgrade`... the only change that was made is that the package was renamed

Reply:
Nvm, I double-checked directly from the dsessler7 repo, I must've been comparing files from the wrong branches or something.